{
	"documenttitle": {
		"text": "Red Hat Security Advisory: python security update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2011:0492"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2011-05-05T14:20:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2011-05-05T18:16:00Z",
		"CurrentReleaseDate": "2011-05-05T14:20:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.0.0",
			"Date": "2021-08-23T04:32:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nUpdated python packages that fix multiple security issues are now available\nfor Red Hat Enterprise Linux 5.\n\nThe Red Hat Security Response Team has rated this update as having moderate\nsecurity impact. Common Vulnerability Scoring System (CVSS) base scores,\nwhich give detailed severity ratings, are available for each vulnerability\nfrom the CVE links in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nPython is an interpreted, interactive, object-oriented programming\nlanguage.\n\nA flaw was found in the Python urllib and urllib2 libraries where they\nwould not differentiate between different target URLs when handling\nautomatic redirects. This caused Python applications using these modules to\nfollow any new URL that they understood, including the \"file://\" URL type.\nThis could allow a remote server to force a local Python application to\nread a local file instead of the remote one, possibly exposing local files\nthat were not meant to be exposed. (CVE-2011-1521)\n\nA race condition was found in the way the Python smtpd module handled new\nconnections. A remote user could use this flaw to cause a Python script\nusing the smtpd module to terminate. (CVE-2010-3493)\n\nAn information disclosure flaw was found in the way the Python\nCGIHTTPServer module processed certain HTTP GET requests. A remote attacker\ncould use a specially-crafted request to obtain the CGI script's source\ncode. (CVE-2011-1015)\n\nA buffer over-read flaw was found in the way the Python Expat parser\nhandled malformed UTF-8 sequences when processing XML files. A\nspecially-crafted XML file could cause Python applications using the Python\nExpat parser to crash while parsing the file. (CVE-2009-3720)\n\nThis update makes Python use the system Expat library rather than its own\ninternal copy; therefore, users must have the version of Expat shipped with\nRHSA-2009:1625 installed, or a later version, to resolve the CVE-2009-3720\nissue.\n\nAll Python users should upgrade to these updated packages, which contain\nbackported patches to correct these issues.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2021 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2011:0492",
				"Description": "https://access.redhat.com/errata/RHSA-2011:0492"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "expat: buffer over-read and crash on XML with malformed UTF-8 sequences",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2009-01-17T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2009-3720",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously-released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/kb/docs/DOC-11259\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2011:0492"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2009-3720",
						"Description": "CVE-2009-3720"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=531697",
						"Description": "bz#531697: CVE-2009-3720 expat: buffer over-read and crash on XML with malformed UTF-8 sequences"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2009-08-21T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"5Client-Workstation:python-0:2.4.3-44.el5",
						"5Client:python-0:2.4.3-44.el5",
						"5Server:python-0:2.4.3-44.el5"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "5.0",
					"Vector": "AV:N/AC:L/Au:N/C:N/I:N/A:P"
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "Python: SMTP proxy RFC 2821 module DoS (uncaught exception) (Issue #9129)",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2010-06-30T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2010-3493",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously-released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/kb/docs/DOC-11259\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2011:0492"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2010-3493",
						"Description": "CVE-2010-3493"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=632200",
						"Description": "bz#632200: CVE-2010-3493 Python: SMTP proxy RFC 2821 module DoS (uncaught exception) (Issue #9129)"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2010-09-08T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"5Client-Workstation:python-0:2.4.3-44.el5",
						"5Client:python-0:2.4.3-44.el5",
						"5Server:python-0:2.4.3-44.el5"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "5.0",
					"Vector": "AV:N/AC:L/Au:N/C:N/I:N/A:P"
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python (CGIHTTPServer): CGI script source code disclosure",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2008-03-07T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2011-1015",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously-released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/kb/docs/DOC-11259\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2011:0492"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2011-1015",
						"Description": "CVE-2011-1015"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=680094",
						"Description": "bz#680094: CVE-2011-1015 python (CGIHTTPServer): CGI script source code disclosure"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2011-02-18T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"5Client-Workstation:python-0:2.4.3-44.el5",
						"5Client:python-0:2.4.3-44.el5",
						"5Server:python-0:2.4.3-44.el5"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "5.0",
					"Vector": "AV:N/AC:L/Au:N/C:P/I:N/A:N"
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python (urllib, urllib2): Improper management of ftp:// and file:// URL schemes (Issue #11662)",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2011-03-24T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2011-1521",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously-released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/kb/docs/DOC-11259\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2011:0492"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2011-1521",
						"Description": "CVE-2011-1521"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=690560",
						"Description": "bz#690560: CVE-2011-1521 python (urllib, urllib2): Improper management of ftp:// and file:// URL schemes (Issue #11662)"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2011-03-24T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"5Client-Workstation:python-0:2.4.3-44.el5",
						"5Client:python-0:2.4.3-44.el5",
						"5Server:python-0:2.4.3-44.el5"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "6.4",
					"Vector": "AV:N/AC:L/Au:N/C:P/I:N/A:P"
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Enterprise Linux",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux Desktop (v. 5 client)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux Desktop (v. 5 client)",
							"productid": "5Client",
							"cpe": "cpe:/o:redhat:enterprise_linux:5::client"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux Desktop Workstation (v. 5 client)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux Desktop Workstation (v. 5 client)",
							"productid": "5Client-Workstation",
							"cpe": "cpe:/o:redhat:enterprise_linux:5::client_workstation"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux (v. 5 server)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux (v. 5 server)",
							"productid": "5Server",
							"cpe": "cpe:/o:redhat:enterprise_linux:5::server"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "python-0:2.4.3-44.el5",
				"fullproductname": {
					"text": "python-2.4.3-44.el5.src.rpm",
					"productid": "python-0:2.4.3-44.el5",
					"cpe": "cpe:/o:redhat:enterprise_linux:5::client"
				}
			}
		],
		"relationship": [
			{
				"productreference": "python-0:2.4.3-44.el5",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "5Client",
				"fullproductname": {
					"text": "python-0:2.4.3-44.el5 as a component of Red Hat Enterprise Linux Desktop (v. 5 client)",
					"productid": "5Client:python-0:2.4.3-44.el5",
					"cpe": "cpe:/o:redhat:enterprise_linux:5::client"
				}
			},
			{
				"productreference": "python-0:2.4.3-44.el5",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "5Client-Workstation",
				"fullproductname": {
					"text": "python-0:2.4.3-44.el5 as a component of Red Hat Enterprise Linux Desktop Workstation (v. 5 client)",
					"productid": "5Client-Workstation:python-0:2.4.3-44.el5",
					"cpe": "cpe:/o:redhat:enterprise_linux:5::client_workstation"
				}
			},
			{
				"productreference": "python-0:2.4.3-44.el5",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "5Server",
				"fullproductname": {
					"text": "python-0:2.4.3-44.el5 as a component of Red Hat Enterprise Linux (v. 5 server)",
					"productid": "5Server:python-0:2.4.3-44.el5",
					"cpe": "cpe:/o:redhat:enterprise_linux:5::server"
				}
			}
		]
	}
}
